src/app/examples/examples.state.ts

```diff
@@ -10,25 +15,54 @@ import { bookReducer } from './crud/books.reducer';
 import { BookState } from './crud/books.model';
 import { formReducer } from './form/form.reducer';
 import { FormState } from './form/form.model';
+import { Injectable } from '@angular/core';
+import { Params, RouterStateSnapshot } from '@angular/router';
 
 export const FEATURE_NAME = 'examples';
 
 export const reducers: ActionReducerMap<ExamplesState> = {
   todos: todosReducer,
   stocks: stockMarketReducer,
   books: bookReducer,
-  form: formReducer
+  form: formReducer,
+  router: routerReducer
 };
 
 export const selectExamples = createFeatureSelector<State, ExamplesState>(
   FEATURE_NAME
 );
+export const selectRouterState = createFeatureSelector<
```

Comment:
This is the reason why the router is undefined, because we're using [`createFeatureSelector`](https://github.com/ngrx/platform/blob/master/docs/store/selectors.md#createfeatureselector).

> The createFeatureSelector is a convenience method for returning a top level feature state. It returns a typed selector function for a feature slice of state.

And in our case the router is nested inside the examples state, to fix this we'll have to use a "simple" createSelector:
```
export const selectRouterState = createSelector(
  selectExamples,
  (state: ExamplesState) => state.router
);
```

But I think it would be better to move the router state to the top level, then it would be possible to use `createFeatureSelector`.

The reason why I would move this to the top level is because this isn't examples bound and is in my eyes the same level as `auth` and `settings`.

Thoughts?

Reply:
Thanks for the explanation and the workaround to get the router state with a selector.

I also thought it makes more sens to get the router state to the top level. So I moved it up.

Now it is nearly working but it should be improved as I need to sync the url with the selectedBookId.

Reply:
I think having a `selectedBookId` isn't needed when we're selecting the book based on the url id.